packages/odie-client/src/data/use-get-support-interactions.ts

```diff
@@ -31,11 +42,18 @@ export const useGetSupportInteractions = (
 				return response.filter( ( interaction ) =>
 					interaction.events.some( ( event ) => event.event_source === provider )
 				);
+			} else if ( ! canConnectToZendesk ) {
+				// When provider is null, we're fetching interactions from all providers.
+				// We need to filter out Zendesk interactions if the user can't connect to Zendesk.
+				return response.filter(
+					( interaction ) =>
+						! interaction.events.some( ( event ) => event.event_source.includes( 'zendesk' ) )
+				);
```

Comment:
If both `provider` and `canConnectToZendesk` are falsy, then we return the `response` without any filtering, right?

Reply:
> If both `provider` and `canConnectToZendesk` are falsy, then we return the response without any filtering, right?

I don't think so, according to the code. When both are falsy, we return interactions from all providers except Zendesk.

Reply:
I meant to say "**Unless I add this condition**, if the provider is `falsy`, it means we'll get interactions from all providers, including Zendesk."